### ISSUE TITLE
mpv grow without stop with vo=opengl+hwdec=vdpau

### ISSUE THREAD

**Issue by Jj0YzL5nvJ** [mpv-player/mpv#3313]
### mpv version and platform

mpv git-6a3da43, Xubuntu 14.04.4 LTS
mpv git-79974e7, Xubuntu 14.04.4 LTS

### Reproduction steps

Playing any video(s) how support hardware acceleration using "mpv --vo=opengl --hwdec=vdpau" with mesa-vdpau-drivers.
Approximately every 3 seconds mpv drain 1 MiB of RAM, with very few stops.

### Expected behavior

Flushing the RAM every that mpv finish reproducing a video. Just like "mpv --vo=opengl --hwdec=no" or "mpv --vo=vdpau --hwdec=vdpau".

### Actual behavior

If mpv play video(s) using "vo=opengl" and "hwdec=vdpau" without stop for several hours. It end devouring the RAM, mpv stop working normally and the system become notoriously slow and start filling the virtual memory.
In short, mpv becomes a resident virus.

### Log files
mpv git-6a3da43: [con_hwdec.txt](https://github.com/mpv-player/mpv/files/350872/con_hwdec.txt)
mpv git-79974e7: [con_hwdec_2.txt](https://github.com/mpv-player/mpv/files/367213/con_hwdec_2.txt)

### Example
![test2](https://cloud.githubusercontent.com/assets/12230570/16894504/4347369c-4b15-11e6-875f-8c5d667a4b23.png)

**Comment by wm4** [comment]
Log?

**Comment by Jj0YzL5nvJ** [comment]
### No hwdec
[sin_hwdec.txt](https://github.com/mpv-player/mpv/files/350864/sin_hwdec.txt)
![sin_hwdec](https://cloud.githubusercontent.com/assets/12230570/16633366/34bb360a-4386-11e6-83a1-d906b8dc95db.png)

### With hwdec/vdpau
[con_hwdec.txt](https://github.com/mpv-player/mpv/files/350872/con_hwdec.txt)
![con_hwdec-1](https://cloud.githubusercontent.com/assets/12230570/16633404/781bf63c-4386-11e6-8df2-2a98faeb8172.png)
![con_hwdec-2](https://cloud.githubusercontent.com/assets/12230570/16633410/7d05d83e-4386-11e6-9c48-96e121689c53.png)
![con_hwdec-3](https://cloud.githubusercontent.com/assets/12230570/16633413/82756442-4386-11e6-8ec7-369048caeab7.png)
![con_hwdec-4](https://cloud.githubusercontent.com/assets/12230570/16633418/86918416-4386-11e6-8df5-8eeffb724771.png)
![con_hwdec-5](https://cloud.githubusercontent.com/assets/12230570/16633421/8af6d538-4386-11e6-80c5-829f0592c100.png)

If you playing many many videos with vdpau, it end devouring all RAM.

**Comment by haasn** [comment]
Sounds like reinitializing vdpau leaks memory.

**Comment by wm4** [comment]
Can't reproduce any of this.

**Comment by Jj0YzL5nvJ** [comment]
I was using ppa:oibaf/graphics-drivers repositories for mesa, vdpau, clang, radeon, etc.
But now it drop my distribution u.u

Thx, I test after a clean install to 16.04 LTS... in the future.
Not nearby future =P

**Comment by Jj0YzL5nvJ** [comment]
Good news! (for me)
I can use vo=vdpau+hwdec=vdpau without problems.
The memory grow occur only with vo=opengl or vo=opengl-hq

Unknown if this info is useful for unmask the true problem.

cya

**Comment by wm4** [comment]
What GPU? (We'd know this if you had posted a log as required.)

**Comment by Jj0YzL5nvJ** [comment]
Juniper PRO [Radeon HD 5750]
In the log you see: GL_RENDERER='Gallium 0.4 on **AMD JUNIPER** (DRM 2.36.0 / 3.13.0-91-generic, LLVM 3.8.0)'
ATI/AMD JUNIPER is the code name...

**Comment by MoSal** [comment]
> GL_VERSION    = 3.0 Mesa 12.1.0-devel (git-d93bacc 2016-06-26 trusty-oibaf-ppa)

As discussed in IRC the other day, a fix for this bug in Mesa was committed in 2016-07-14.
https://cgit.freedesktop.org/mesa/mesa/commit/?id=9ce52baf7f479d506bcea17a371dcf2fad92f139

**Comment by Jj0YzL5nvJ** [comment]
The symptoms already are not present with Xubuntu 16.04.1 LTS and mpv git-0432ab8.
Apparently is like @MoSal said, this was a bug in Mesa. I use Padoka PPA now,
